blockchain/validators/src/metadata/mod.rs

```diff
@@ -2,128 +2,80 @@
 
 //! A [BlockMetadata] validator.
 
+pub mod avr;
 pub mod key_range;
 
 use crate::{ParseError, ValidationError};
+use avr::AvrValidator;
 use key_range::KeyRangeValidator;
-use mc_blockchain_types::{BlockIndex, BlockMetadata};
+use mc_blockchain_types::{BlockData, BlockIndex, BlockMetadata};
+use mc_common::ResponderId;
 use std::path::Path;
 
 /// A [BlockMetadata] validator that validates metadata signatures, signing
 /// keys, and AVRs.
 pub struct MetadataValidator {
-    key_range: KeyRangeValidator,
+    /// Message signing key & metadata signature validator
+    pub key_range: KeyRangeValidator,
+
+    /// AVR & block signing key validator
+    pub avr: AvrValidator,
 }
 
 impl MetadataValidator {
     /// Instantiate a validator.
     ///
     /// Args:
     /// `signers_config`: Path to `metadata-signers.toml`
-    pub fn new(signers_config: impl AsRef<Path>) -> Result<Self, ParseError> {
+    pub fn new(
+        signers_config: impl AsRef<Path>,
+        avr_history: impl AsRef<Path>,
+    ) -> Result<Self, ParseError> {
         let key_range = KeyRangeValidator::load(signers_config)?;
-
-        Ok(Self { key_range })
+        let avr = AvrValidator::load(avr_history)?;
+        Ok(Self { key_range, avr })
     }
 
-    /// Validate that the given metadata is valid at the given block index.
+    /// Validate block metadata signature, signing keys, and AVRs
+    ///
+    /// Args:
```

Comment:
Formatting isn't rustdoc compatible - see https://doc.rust-lang.org/rust-by-example/meta/doc.html for an example